samcli/local/apigw/local_apigw_service.py

```diff
@@ -315,6 +284,77 @@ def _construct_error_handling(self):
         # Something went wrong
         self._app.register_error_handler(500, ServiceErrorResponses.lambda_failure_response)
 
+    def _valid_identity_sources(self, route: Route) -> bool:
+        """
+        Validates if the route contains all the valid identity sources defined in the route's Lambda Authorizer
+
+        Parameters
+        ----------
+        route: Route
+            the Route object that contains the Lambda Authorizer definition
+
+        Returns
+        -------
+        bool
+            true if all the identity sources are present and valid
+        """
+        lambda_auth = route.authorizer_object
+
+        if not isinstance(lambda_auth, LambdaAuthorizer):
+            return False
+
+        identity_sources = lambda_auth.identity_sources
+
+        endpoint = PathConverter.convert_path_to_api_gateway(request.endpoint)
+        method = request.method
+
+        if lambda_auth.payload_version == LambdaAuthorizer.PAYLOAD_V1:
```

Comment:
the if else could just be two functions, esp if one branch is v1 and other is v2.